Update STM32MP15 Device Tree files to Linux 5.19-rc6

jneuhauser opened the pull request:
<!--
    If you are new to submitting pull requests to OP-TEE, then please have a
    look at the list below and tick them off before submitting the pull request.

    1. Read our contribution guidelines:
         https://optee.readthedocs.io/en/latest/general/contribute.html

    2. Read the contribution section in Notice.md and pay extra attention to the
       "Developer Certificate of Origin" in the contribution guidelines.

    3. You should run checkpatch preferably before submitting the pull request.

    4. When everything has been reviewed, you will need to squash, rebase and
       add tags like `Reviewed-by`, `Acked-by`, `Tested-by` etc. More details
       about this can also be found on the link provided above.

    NOTE: This comment will not be shown in the pull request, so no harm keeping
    it, but feel free to remove it if you like.
-->
Syncronize with STM32MP15 Device Tree files from Linux v5.19-rc6.

I've readded the etzpc node, disabled the rcc secure-status and removed the ressources related GPLv2 licensing model for all boards like in the previous bump of the stm32mp15 dts(i) files.

This update is required to add a new boards based on Linux 5.19-rc6
dts file.

Ping @etienne-lms as the STM32MP1 platform maintainer.

jneuhauser commented:
We have tested the device tree bump only with our DHSOM boards, whose current state can be found in my repo:
https://github.com/jneuhauser/optee_os/commits/dev/stm32mp1-dhsom

I can provide bootlogs from the OP-TEE developer setup. `xtest` runs all test successfully.

jneuhauser commented:
I've dropped the new added cryp1 status okay from dk2 and ed1 boards as adding it on our borads results in the following `panic()`:
```
E/TC:0 0 check_rcc_secure_configuration:630 RCC non-secure MCKPROT not set and CRYP1 (14) secure
E/TC:0 0 Panic
E/TC:0 0 TEE load address @ 0x2ffc0000
E/TC:0 0 Call stack:
E/TC:0 0  0x2ffc1a99 print_kernel_stack at core/arch/arm/kernel/unwind_arm32.c:109
E/TC:0 0  0x2ffc55b5 __do_panic at core/kernel/panic.c:31
E/TC:0 0  0x3001c0c9 check_rcc_secure_configuration at core/arch/arm/plat-stm32mp1/shared_resources.c:623
...
```

Also readd the interrupt definitions for iwdg2 as added in https://github.com/OP-TEE/optee_os/commit/81ed3bce8c10b7df1406d476aa4ffba1db0fbd6a.

I'll mark this PR as RFC as long a got some feedback from @etienne-lms

etienne-lms reviewed (commented):
Hello @jneuhauser,
Late feedback. Sorry.
The P-R looks almost good to me (1 comment on TAMP node). I think we should explicitly disable the nodes OP-TEE is not expected to use in the supported board DTS files. But that can be done later with a specific P-R.

For commit "dt-bindings: gpio: add GPIO_PULL_{UP,DOWN} definitions":
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>`

For commit "dts: stm32mp15: bump to Linux v5.19-rc6 dts files":
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>` with comment addressed.

jneuhauser commented:
I have addressed your comment regarding the clock property in the tamp node and rebased the PR on master.

Let me know if I can do any furter work or provide some logs from our STM32MP15 based DHSOM to get this merged.

etienne-lms reviewed (approved):
Thanks @jneuhauser for the update. Looks all good to me. CI is complaining but I think it's out of the scope of your changes and they'll likely get merged during summer.

jneuhauser commented:
The CI tests were successful before the minor update. It looks like something is wrong with CI in general or it has something to do with rebase to master.

Thank you @etienne-lms for your time.

jforissier commented:
Indeed the CI failures are unrelated, I will post a PR soon to address the issues. Thanks!
